tensorflow_addons/activations/tanhshrink.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+from tensorflow_addons.utils.resource_loader import get_path_to_datafile
+
+_activation_ops_so = tf.load_op_library(
+    get_path_to_datafile("custom_ops/activations/_activation_ops.so"))
+
+
+@keras_utils.register_keras_custom_object
+@tf.function
+def tanhshrink(features):
+    """Applies the element-wise function: x - tanh(x)
+
+    Args:
+        features: A `Tensor`. Must be one of the following types:
+            `float16`, `float32`, `float64`.
+    Returns:
+        A `Tensor`. Has the same type as `features`.
+    """
+    with tf.name_scope("tanhshrink") as name:
+        features = tf.convert_to_tensor(features, name="features")
+        return _activation_ops_so.addons_tanhshrink(features, name=name)
```

Comment:
Do we need this name scope?

Reply:
Perhaps we can remove it as there are two lines code here. What do you think, @fsx950223 and @WindQAQ ?

Reply:
addons_tanhshrink is an independent c++ operation.